src/parsers.ts

```diff
@@ -0,0 +1,153 @@
+    isUint(data) && data <= maxSize
+
+export const parseInt = (data: unknown, errMsg: ParseErrorReason): Int => {
+    validate(isInt(data), errMsg)
+    return data
+}
+
+export const parseUint = (data: unknown, errMsg: ParseErrorReason): Uint => {
+    validate(isUint(data), errMsg)
+    return data
+}
+
+export const parseStringOfMaxLength = <L extends number>(data: unknown, maxLength: L, errMsg: ParseErrorReason): MaxlenString<L> => {
+    validate(isStringOfMaxLength(data, maxLength), errMsg)
+    return data
+}
+
+export const parseBuffer = (data: unknown, errMsg: ParseErrorReason): Buffer => {
+    validate(isBuffer(data), errMsg)
+    return data
+}
+
+export const parseBufferOfLength = <L extends number>(data: unknown, length: L, errMsg: ParseErrorReason): FixlenBuffer<L> => {
+    validate(isBufferOfLength(data, length), errMsg)
+    return data
+}
+
+export const parseBufferOfMaxLength = <L extends number>(data: unknown, maxLength: L, errMsg: ParseErrorReason): MaxlenBuffer<L> => {
+    validate(isBufferOfMaxLength(data, maxLength), errMsg)
+    return data
+}
+
+/**
+ * Defines a generic parser that takes only the data, it assumes other
+ * arguments of the parser are already bound.
+ * @see createParser
+ */
+export type Parser<T> = (data: unknown) => T
+
+/**
+ * Takes a parsing function and binds its arguments except for the data
+ * returns a parser function that only requires the data to execute
+ * @see Parser
+ */
+export function createParser<L extends number>(paser: (data: unknown, length: L, errMsg: ParseErrorReason) => FixlenBuffer<L>, length: L, errMsg: ParseErrorReason): Parser<FixlenBuffer<L>>
+export function createParser<L extends number>(paser: (data: unknown, maxLength: L, errMsg: ParseErrorReason) => MaxlenBuffer<L>, maxLength: L, errMsg: ParseErrorReason): Parser<MaxlenBuffer<L>>
+export function createParser<L extends number>(paser: (data: unknown, maxLength: L, errMsg: ParseErrorReason) => MaxlenString<L>, maxLength: L, errMsg: ParseErrorReason): Parser<MaxlenString<L>>
+export function createParser<T, A extends any[]>(parser: (data: unknown, ...args: [...A]) => T, ...args: [...A]): Parser<T>
+export function createParser<T, A extends any[]>(parser: (data: unknown, ...args: [...A]) => T, ...args: [...A]): Parser<T> {
+    return (data: unknown) => parser(data, ...args)
+}
+
+export const parseMap = <K, V>(
+    data: unknown,
+    parseKey: Parser<K>,
+    parseValue: Parser<V>,
+    errMsg: ParseErrorReason,
+): Map<K, V> => {
+    validate(isMap(data), errMsg)
+    return new Map(Array.from(data.entries()).map(([key, value]) => [parseKey(key), parseValue(value)]))
+}
+
+export const parseArray = <T>(
+    data: unknown,
+    parseEntry: Parser<T>,
+    errMsg: ParseErrorReason,
+): Array<T> => {
+    validate(isArray(data), errMsg)
+    return data.map((value) => parseEntry(value))
+}
+
+/**
+ * Defines a tuple of parsers each for the individual type from tuple `T`
```

Comment:
It's a tuple, because of `T extends any[]`, so `T` is an array of `any` elements and in `{ [K in keyof T]: Parser<T[K]> }` the `K` is an array index. But seeing as this `Parsers<T>` type is only used in the `parseTuple` function I think I can inline this type.